### DISCUSSION TITLE
Using REPL without build

### DISCUSSION THREAD

**Discussion by dvd101x** [josdejong/mathjs#3145] [Q&A] [answered]
When doing changes to the source code the REPL doesn't see the changes until I build. The documentation states that REPL doesn't require to build on every little change. I would like to know if I'm doing something wrong.

# Steps to reproduce

1. Make a change in the source files
2. Run `./bin/repl.js`
3. See my change is not implemented
4. Exit REPL
5. Run `npm run build`
6. Run `./bin/repl.js`
7. See my change implemented

> For debugging purposes, bin/repl.js provides a REPL (Read Evaluate Print Loop) for interactive testing of mathjs without having to build new build files after every little change to the source files. You can either start it directly (./bin/repl.js) or via node (node bin/repl.js). https://mathjs.org/docs/command_line_interface.html#command-line-debugging-repl

**Comment by kungfooman** [comment]
Looks like it's resolving a build package from this commit on: https://github.com/josdejong/mathjs/commit/6f007157547c02ab6bfda06842bb1e118adbb9e8

**Comment by josdejong** [comment, marked as answer]
Good point. The REPL was created in 2015 (9 years ago), and was changed to use the build in 2020 (4 years ago). At that time we should have updated the docs accordingly, since we now need a build step. 

For now I've updated the documentation about the REPL. It would be nice to be able to use the REPL (or the library itself) without build step, but I think that is not possible right now.

**Reply by josdejong** [reply]
See 5fb909ab5906887f2edca2734288b7ed9bf254cf

**Reply by josdejong** [reply]
The docs are updated now in `v12.3.1`

**Reply by dvd101x** [reply]
Thank you Jos!
